### PR TITLE
Configure check for Windows

### PR DESCRIPTION
**Pull request by Sgitario**
Added windows check.
WARNING: The check is not passing.

### CODE REVIEW

**Comment by Sgitario** [comment]
@rsvoboda I have checked that the new windows check is not passing. Do you want me to create a separate task for fixing?

**Comment by Sgitario** [comment]
@rsvoboda PR updated. Now the artifacts are being attached with the logs from the failures.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -26,15 +26,17 @@ jobs:
...
       - name: Zip Artifacts
+        if: failure()
         run: |
-          zip -r artifacts-jvm${{ matrix.java }}.zip . -i '*-reports/*' '*/archived-logs/*'
+          zip -r linux-build-released-jvm${{ matrix.java }}.zip . -i '*-reports/*' '*/archived-logs/*'
       - name: Archive artifacts
         uses: actions/upload-artifact@v1
+        if: failure()
         with:
           name: ci-artifacts
```
Comment (rsvoboda) [inline .github/workflows/ci.yaml:36]: this should be adjusted to reflect to which build the artifact belongs

atm there is 4x `name: ci-artifacts`

Comment (Sgitario) [reply .github/workflows/ci.yaml:36]: This configuration results in a single ci-artifact with up to four zip files (one per check). 
I can configure to have four artifacts, each artifact with one zip file, but I think it's better to download one file and extract all the check outputs at once ...

Comment (rsvoboda) [reply .github/workflows/ci.yaml:36]: aha, ok, I thought it would get overridden

**Comment by rsvoboda** [comment]
Please add "-e" to https://github.com/quarkus-qe/quarkus-startstop/blob/master/testsuite/src/it/java/io/quarkus/ts/startstop/utils/MvnCmds.java#L39 so we see stacktrace of failed dev mode command

File: testsuite/src/it/java/io/quarkus/ts/startstop/utils/Logs.java [hunk 2]
```diff
@@ -84,13 +84,20 @@ public static void checkLog(String testClass, String testMethod, Apps app, MvnCm
...
                     } else {
                         offendingLines.add(line);
                     }
                 }
             }
+            
+            // Quarkus is not being gratefully shutdown in Windows when running in Dev mode.
+            // Reported by https://github.com/quarkusio/quarkus/issues/14647.
+            if (Commands.isThisWindows && offendingLines.stream().anyMatch(line -> WINDOWS_DEV_MODE_ERROR.matcher(line).matches())) {
+            	offendingLines.clear();
```
Comment (rsvoboda) [inline testsuite/src/it/java/io/quarkus/ts/startstop/utils/Logs.java:98]: this clear all the errors / warnings.

You should remove just the one related to shutdown hook not being processed properly => exit code is 1

File: app-full-microprofile/threshold.properties [hunk 3]
```diff
@@ -2,5 +2,5 @@ linux.jvm.time.to.first.ok.request.threshold.ms=2600
 linux.jvm.RSS.threshold.kB=550000
 linux.native.time.to.first.ok.request.threshold.ms=50
 linux.native.RSS.threshold.kB=120000
```
Comment (rsvoboda) [inline app-full-microprofile/threshold.properties:4]: Thresholds can be defined on command line too - https://github.com/quarkus-qe/quarkus-startstop#thresholds

We do not activelly run on Windows, so making direct adjustments seems OK for now

File: testsuite/src/it/java/io/quarkus/ts/startstop/SpecialCharsTest.java [hunk 4]
```diff
@@ -202,6 +202,7 @@ public void diacriticsJVM(TestInfo testInfo) throws IOException, InterruptedExce
     }
 
     @Test
+    @DisabledOnOs({OS.WINDOWS}) // https://github.com/quarkusio/quarkus/issues/9707
```
Comment (rsvoboda) [inline testsuite/src/it/java/io/quarkus/ts/startstop/SpecialCharsTest.java:205]: good point, I think to make this work you need to have Czech language support installed on Windows, so by default this test doesn't work.

File: testsuite/src/test/java/io/quarkus/ts/startstop/ProcessTest.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.IOException;
+
+import org.junit.jupiter.api.Test;
+
+public class ProcessTest {
+
+    @Test
+    public void dummyTest() throws IOException, InterruptedException {
```
Comment (rsvoboda) [inline testsuite/src/test/java/io/quarkus/ts/startstop/ProcessTest.java]: give it better name

and why is it included?

**Review by rsvoboda — CHANGES_REQUESTED** [review]

File: testsuite/src/test/java/io/quarkus/ts/startstop/ProcessTest.java (hunk 5 above)
Comment (Sgitario) [reply testsuite/src/test/java/io/quarkus/ts/startstop/ProcessTest.java]: This was a leftover, I had removed this test already.

**Comment by Sgitario** [comment]
This will be done as part of https://github.com/quarkus-qe/quarkus-startstop/pull/75. Closing.

**Pull request closed without merging**